python/flask/app-run-security-config.yaml

```diff
@@ -0,0 +1,53 @@
+
+# Flask check3: Flask application app.run best practices https://flask.palletsprojects.com/en/1.1.x/api/?highlight=run#flask.Flask.run
+rules:
+  # # . Rules for app.run() scope
+  - id: app_run_with_guard
+    pattern: |
+      if $X == '__main__':
```

Comment:
So this works in the end? I remember you had problems with that pattern.

Reply:
I think it worked in the sense that it did not fire on 

```
if __name__ == "__main__":
    app.run()
```
It could have been because the pattern did not work OR because I had "OK" for the rule. I will double check.

Reply:
Double checks this works on sgrep as of `[3e03273dff322819674670051d374de6a638c74f](https://github.com/returntocorp/sgrep/commit/3e03273dff322819674670051d374de6a638c74f)`.
```json
ulziibayarotgonbaatar@ulziibayars-MacBook-Pro sgrep % ./_build/default/bin/main_sgrep.exe -sgrep_lint -rule_file ~/Workspace/sgrep-rules/python/flask/app-run-security-config.yaml ~/Workspace/sgrep-rules/tests/python/flask/app_run_security_config
{
  "results": [
    {
      "check_id": "sgrep-lint-<avoid_using_app_run_directly>",
      "path": "test_avoid_using_app_run_directly.py",
      "start": { "line": 4, "col": 1 },
      "end": { "line": 4, "col": 4 },
      "extra": {
        "message":
          "app.run(...) is ignored by flask. Consider putting app.run(...) behind a guard",
        "line": "NO LINE"
      }
    },
    {
      "check_id": "sgrep-lint-<avoid_using_app_run_directly_no_option>",
      "path": "test_avoid_using_app_run_directly.py",
      "start": { "line": 1, "col": 1 },
      "end": { "line": 1, "col": 4 },
      "extra": {
        "message":
          "app.run() is ignored by flask. Consider putting app.run() behind a guard",
        "line": "EMPTY FILE"
      }
    }
  ]
}
```